features/topology-map/plugins/org.opennms.features.topology.plugins.topo.application/src/main/java/org/opennms/features/topology/plugins/topo/application/browsers/ApplicationSelectionLinkGenerator.java

```diff
@@ -67,16 +67,21 @@ public Object generateCell(final Table source, final Object itemId, Object colum
 				button.addClickListener(new Button.ClickListener() {
 					@Override
 					public void buttonClick(Button.ClickEvent event) {
-							ApplicationCriteria applicationCriteria = graphContainer.findSingleCriteria(ApplicationCriteria.class);
-							if (applicationCriteria == null) {
-								applicationCriteria = new ApplicationCriteria();
-								graphContainer.addCriteria(applicationCriteria);
-							}
-							if (!Objects.equals(applicationCriteria.getApplicationId(), String.valueOf(idProperty.getValue()))) {
-								applicationCriteria.setApplicationId(String.valueOf(idProperty.getValue()));
-								graphContainer.setDirty(true);
-								graphContainer.redoLayout();
-							}
+                        UI ui = UI.getCurrent();
```

Comment:
Can you please add a comment why you are doing this instead of using the graphContainer.